### PR TITLE
Use any dolibarr enabled paiement mode in TakePOS

### PR DESCRIPTION
**Pull request by dbeniamine**
# New Use any dolibarr enabled paiement mode

With this commit, one can use any paiement mode enabled in Dolibarr `llc_c_paiment` when validating a bill from TakePOS.

To enable a paiment mode in TakePOS, it must be enabled in Dolibarr AND a bank account must be associated to the paiment mode in TakePOS setting page.

### CODE REVIEW

File: htdocs/takepos/admin/setup.php [hunk 1]
```diff
@@ -41,19 +41,30 @@
 
 $langs->loadLangs(array("admin", "cashdesk"));
 
+global $db;
+
+$sql="SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement WHERE active=1 ORDER BY libelle";
+$resql = $db->query($sql);
```
Comment (hregis) [inline htdocs/takepos/admin/setup.php:50]: @dbeniamine 
for compatibility with multicompany : 

```
$sql = "SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement";
$sql.= " WHERE entity IN (".getEntity('c_paiement').")";
$sql.= " AND active = 1";
$sql.= " ORDER BY libelle";
```

File: htdocs/takepos/invoice.php [hunk 2]
```diff
@@ -40,6 +40,10 @@
 $idline = GETPOST('idline');
 $desc = GETPOST('desc', 'alpha');
 $pay = GETPOST('pay');
+$sql="SELECT rowid FROM ".MAIN_DB_PREFIX."facture where ref='(PROV-POS-".$place.")'";
+$resql = $db->query($sql);
```
Comment (hregis) [inline htdocs/takepos/invoice.php]: @dbeniamine 
for multicompany compatibility : 

```
$sql = "SELECT rowid FROM ".MAIN_DB_PREFIX."facture";
$sql.= " WHERE entity IN "(".getEntity('takeposinvoice').")";
$sql.= " AND ref='(PROV-POS-".$place.")'";
```

File: htdocs/takepos/invoice.php [hunk 3]
```diff
@@ -48,16 +52,20 @@
 if ($ret > 0) $placeid = $invoice->id;
 
 
+// Retrieve paiementid
+$sql="SELECT id FROM ".MAIN_DB_PREFIX."c_paiement WHERE code='$pay'";
+$resql = $db->query($sql);
```
Comment (hregis) [inline htdocs/takepos/invoice.php:59]: @dbeniamine 
for compatibility with multicompany :

```
$sql = "SELECT id FROM ".MAIN_DB_PREFIX."c_paiement";
$sql.= " WHERE entity IN (".getEntity('c_paiement').")";
$sql.= " AND code = '$pay'";
```

File: htdocs/takepos/pay.php [hunk 4]
```diff
@@ -50,6 +50,16 @@
 top_htmlhead($head, $title, $disablejs, $disablehead, $arrayofjs, $arrayofcss);
 
 $langs->loadLangs(array("main", "bills", "cashdesk"));
+
+$sql="SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement WHERE active=1 ORDER BY libelle";
+$resql = $db->query($sql);
```
Comment (hregis) [inline htdocs/takepos/pay.php:58]: @dbeniamine 
for compatibility with multicompany :

```
$sql = "SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement";
$sql.= " WHERE entity IN (".getEntity('c_paiement').")";
$sql.= " AND active = 1";
$sql.= " ORDER BY libelle";
```

File: htdocs/takepos/takepos.php [hunk 5]
```diff
@@ -450,7 +450,17 @@ function MoreActions(totalactions){
 
 <?php
 // TakePOS setup check
-if (empty($conf->global->CASHDESK_ID_THIRDPARTY) or empty($conf->global->CASHDESK_ID_BANKACCOUNT_CASH) or empty($conf->global->CASHDESK_ID_BANKACCOUNT_CB)) {
+// TODO
+$sql="SELECT code,libelle FROM ".MAIN_DB_PREFIX."c_paiement WHERE active=1 ORDER BY libelle";
+$resql = $db->query($sql);
```
Comment (hregis) [inline htdocs/takepos/takepos.php:458]: @dbeniamine 
for compatibility with multicompany :

```
$sql = "SELECT code, libelle FROM ".MAIN_DB_PREFIX."c_paiement";
$sql.= " WHERE entity IN (".getEntity('c_paiement').")";
$sql.= " AND active = 1";
$sql.= " ORDER BY libelle";
```

File: htdocs/takepos/admin/setup.php (hunk 1 above)
Comment (dbeniamine) [reply htdocs/takepos/admin/setup.php:50]: done in 32303bbe4827cba26e8b57b5d17844766a1fc2c9

File: htdocs/takepos/invoice.php (hunk 3 above)
Comment (dbeniamine) [reply htdocs/takepos/invoice.php:59]: done in 32303bbe4827cba26e8b57b5d17844766a1fc2c9

File: htdocs/takepos/pay.php (hunk 4 above)
Comment (dbeniamine) [reply htdocs/takepos/pay.php:58]: done in 32303bbe4827cba26e8b57b5d17844766a1fc2c9

File: htdocs/takepos/takepos.php (hunk 5 above)
Comment (dbeniamine) [reply htdocs/takepos/takepos.php:458]: done in 32303bbe4827cba26e8b57b5d17844766a1fc2c9

File: htdocs/takepos/invoice.php (hunk 2 above)
Comment (dbeniamine) [reply htdocs/takepos/invoice.php]: This sql request was dead code revived by myself as I ported my contribution to TakePOS9, I removed it.

**Pull request merged**